hphp/runtime/ext/datetime/ext_datetime.php

```diff
@@ -876,8 +875,21 @@ function timezone_name_get(DateTimeZone $timezone): string {
 }
 
 <<__ParamCoerceModeFalse>>
-function timezone_open(string $timezone): DateTimeZone {
-  return new DateTimeZone($timezone);
+function timezone_open(string $timezone): mixed {
+  try {
+    return new DateTimeZone($timezone);
+  }
+  catch (Exception $e) {
+    trigger_error(
+      str_replace(
+        'DateTimeZone::__construct():',
+        __FUNCTION__ . '():',
+        $e->getMessage()
+      ),
+      E_WARNING
```

Comment:
By the way, trigger_error() does not allow anything except E_USER_* constants for the second parameter, but I noticed that in your code you have plenty of calls that use E_WARNING; is there some kind of exception for these particular files?